js/index.js

```diff
@@ -43,26 +43,81 @@ function renderPepperoni() {
 }
 
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
-}
+ document.querySelectorAll('.mushroom').forEach(onMush =>{
+   if (state.mushrooms) {
+     onMush.style.visibility = "visible";
+    }  else {
+       onMush.style.visibility = "hidden";
+     }
+   });
+ }
+// Iteration 1: set the visibility of `<section class="mushroom">`
+
 
 function renderGreenPeppers() {
+  document.querySelectorAll('.green-pepper').forEach(onGreenPap => {
+    if (state.greenPeppers){
+      onGreenPap.style.visibility = 'visible';
+    } else {
+      onGreenPap.style.visibility = 'hidden';
+    }
+  });
   // Iteration 1: set the visibility of `<section class="green-pepper">`
 }
 
 function renderWhiteSauce() {
+  const whiteSauce = document.querySelector('.sauce');
+if(state.whiteSauce){
+  whiteSauce.classList.add('sauce-white')
+}
+else {
+  whiteSauce.classList.remove('sauce-white');
+}
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
 }
 
 function renderGlutenFreeCrust() {
+  const glutenFreeCrust = document.querySelector(".crust");
+    if(state.glutenFreeCrust){
+      glutenFreeCrust.classList.add('crust-gluten-free');
+
+    }
+else {
+  glutenFreeCrust.classList.remove('crust-gluten-free');
+}  
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
 }
 
 function renderButtons() {
+const buttons = document.querySelectorAll('.panel.control');
+buttons.toggleClass('active');
+
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
 }
```

Comment:
This is a solution example for this function:

`function renderButtons() { var ingArr = Object.keys(state); document.querySelectorAll(".btn").forEach(function(item, index) { if (state[ingArr[index]]) { item.classList.add("active"); } else { item.classList.remove("active"); } }); }`